Writerside/topics/raptor-xx.md

```diff
@@ -360,48 +347,16 @@ helping you identify and resolve performance bottlenecks and resource management
 ## Foreign Function and Memory (FFM) API
 
 Initially, we explored the possibility of calling the C++ RAPTOR implementation from Java using the Foreign Function and
-Memory (FFM) API. However, due to the complexity of mapping intricate data structures between C++ and Java, we
-ultimately decided not to pursue this implementation.
-
-### Challenges and C Compatibility
-
-Integrating C++ code with Java through the FFM API presents several technical challenges. Below are the key difficulties
-we encountered:
-
-#### 1. Memory Management
-
-Managing off-heap memory in Java can be challenging when working with native code. The FFM API offers the
-`MemorySegment` and `Arena` classes to facilitate memory management, allowing developers to allocate and deallocate
-native memory efficiently. However, improper handling of this off-heap memory can lead to memory leaks or crashes.
-Developers need to ensure memory management practices to avoid such issues.
-
-#### 2. Data Conversion
-
-Converting data between Java and native C types is another significant hurdle. The FFM API provides `ValueLayout`
-classes to define the memory layout of data types in both Java and native code. However, ensuring that these layouts
-match the exact structure of the native code (C++) requires careful attention. In complex data structures, especially
-those with pointers or custom layouts, mismatches can result in runtime errors or undefined behavior.
-
-#### 3. Error Handling
-
-Error handling becomes more difficult when working with native code. Bugs or faults in C++ code can cause the JVM to
-crash, making debugging a more complicated process. While Java provides robust error handling, diagnosing issues in
-native code may require external tools such as native debuggers, core dumps, and more intricate diagnostic setups. This
-introduces additional overhead during development and testing.
-
-#### 4. Platform Differences
-
-Although the FFM API abstracts some platform-specific details, developers must still account for differences in calling
-conventions and memory layouts between platforms (e.g., Windows vs. Linux). These differences can introduce subtle bugs
-or performance issues when deploying the same Java code across different operating systems. Cross-platform compatibility
-must be thoroughly tested to avoid such pitfalls.
+Memory (FFM) API. However, we faced significant challenges due to the complexity of mapping intricate data structures
+between C++ and Java. Ultimately, we decided against this approach.
 
-### Conclusion (FFM)
+The FFM API was not well-suited for our application. Sending routing requests from Java to C++ required C++ to operate
```

Comment:
Suggestion:

The FFM (Foreign Function & Memory) API wasn't well-suited for our application. Sending routing requests from Java to C++ required C++ to interact with a memory structure built in Java, which would have been complex to implement. Since Java was the primary application, the alternative was rebuilding the schedule from files in C++ for each request, which would have been extremely inefficient.

A more effective approach would have been to implement a dedicated C++ service that Java could call for routing tasks. The FFM API, while useful for certain types of function replacements, wasn't designed to manage the kind of larger, integrated workflow we needed, making it unsuitable for our use case.